payroll/search.py

```diff
@@ -75,6 +73,33 @@ def _format_results(cls, results):
         return results
 
 
+class LazyPaginatedResults(list):
```

Comment:
since we are not really using any elements of the list, I think it would be clearer to have this be 

```
class LazyPaginatedResult(colllections.abc.Sequence):
```

https://docs.python.org/3/library/collections.abc.html#collections.abc.Sequence